cmb2-conditionals.php

```diff
@@ -83,13 +83,10 @@ class CMB2_Conditionals {
 		/**
 		 * Constructor - Set up the actions for the plugin.
 		 */
-		public function __construct() {
-			if ( ! defined( 'CMB2_LOADED' ) || false === CMB2_LOADED ) {
-				return;
-			}
+		public function __construct() {			
 
-			add_action( 'admin_init', array( $this, 'admin_init' ), self::PRIORITY );
-			add_action( 'admin_footer', array( $this, 'admin_footer' ), self::PRIORITY );
+			add_action( 'cmb2_admin_init', array( $this, 'admin_init' ), self::PRIORITY );
+			add_action( 'cmb2_footer_enqueue', array( $this, 'admin_footer' ), self::PRIORITY );
 
```

Comment:
> Looks like cmb2_admin_init was added August 30, 2015, during version 2.1.0. cmb2_footer_enqueue was added February 15, 2017 during version 2.2.3.1. 

For the `cmb2_admin_init` hook, I'm ok with moving up the minimum CMB2 requirement as that version is nearly two years old.
For the `cmb2_footer_enqueue` hook, I don't think that's a good idea as it's a too recent change, so I would advocate either not making the change or wrapping it in a version check.

> The problem with checking the version of CMB2 before adding these hooks is that often times this plugin could be loaded before CMB is, hence the point of the fork to begin with. I'll clearly annotate the requirement and resubmit.

That's what the `CMB2_LOADED` check a couple of lines earlier is about and this should not be removed.
If you want to influence the running order of which plugin is loaded when, you should change the priority of the `add_action( 'plugins_loaded', 'cmb2_conditionals_init' );` call in line 256. It now defaults to 10. Changing that to - for instance - `20` could already solve your issue.

If it doesn't, I would suggest adding an additional if/else within the conditional around that line to also check `did_action( 'plugins_loaded' )`.
```php
if ( ( function_exists( 'wp_installing' ) && wp_installing() === false ) || ( ! function_exists( 'wp_installing' ) && ( ! defined( 'WP_INSTALLING' ) || WP_INSTALLING === false ) ) ) {
	if ( did_action( 'plugins_loaded' ) > 0 ) {
		cmb2_conditionals_init();
	} else {
		add_action( 'plugins_loaded', 'cmb2_conditionals_init' );
	}
}
```


> Are there any other areas it should?

`composer.json` comes to mind... though a search of the files in the repo might turn up more places.